libs-scala/scala-utils/src/main/scala/com/daml/scalautil/SomeArrayEquals.scala

```diff
@@ -51,7 +51,7 @@ trait SomeArrayEquals extends Product with Serializable {
 }
 
 object SomeArrayEquals {
```

Comment:
I really not sure this should be move outside speedy.
This deals only with Array at the first level, if one of the argument of your `Product` is of type `List[Array[_]]` or `Option[Array[_]]`, this latter will be compare with standard equality.

Reply:
for instance the following test will succeed :

```
"equals" should {
    "distinguish nested Arrays" in {
      case class A(x: Option[Array[Int]])

      val a = A(Some(Array(1)))
      val b = A(Some(Array(1)))

      a shouldBe a
      a shouldNot be(b)
    }
```